halpybot/commands/caseutils.py

```diff
@@ -58,41 +97,555 @@ async def cmd_go(ctx: Context, args: List[str]):
+    Aliases: n/a
+    """
+    try:
+        await ctx.bot.board.rename_case(args[1], case, ctx.sender)
+    except (AssertionError, ValueError) as err:
+        return await ctx.reply(str(err))
+    for channel in config.channels.rescue_channels:
+        await ctx.bot.message(
+            channel,
+            f"Client for case {case.board_id} set to {args[1]!r} from {case.client_name!r}",
+        )
+
+
+@Commands.command("ircn")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_ircn(ctx: Context, args: List[str], case: Case):
+    """
+    Rename the user of an active case
+
+    Usage: !ircn [board ID] [valid IRC user]
+    Aliases: n/a
+    """
+    try:
+        await User.get_info(ctx.bot, args[1])
+    except (AttributeError, NoUserFound):
+        return await ctx.reply("That's not an IRC user!")
+    update = await update_single_elem_case_prep(
+        ctx=ctx, case=case, action="IRC Name", new_key="irc_nick", new_item=args[1]
+    )
+    if update:
+        return await ctx.reply(update)
+
+
+@Commands.command("system")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_system(ctx: Context, args: List[str], case: Case):
+    """
+    Change the system of an active case
+
+    Usage: !rename [board ID] [new system]
+    Aliases: n/a
+    """
+    newsys: str = " ".join(args[1:])
+    newsys = await sys_cleaner(newsys)
+    update = await update_single_elem_case_prep(
+        ctx=ctx, case=case, action="Client System", new_key="system", new_item=newsys
+    )
+    if update:
+        await ctx.reply(update)
+    try:
+        landmark, distance, direction = await checklandmarks(newsys)
+    except (NoResultsEDSM, EDSMConnectionError):
+        return await ctx.reply(
+            f"{newsys} not found in EDSM or unable to connect to EDSM."
+        )
+    return await ctx.reply(
+        f"The closest landmark system is {landmark}, {distance} LY {direction} of {newsys}."
+    )
+
+
+@Commands.command("status")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_status(ctx: Context, args: List[str], case: Case):
+    """
+    Change the activity status of a case
+
+    Usage: !status [board ID] [new status]
+    Aliases: n/a
+    """
     try:
-        case: Case = await get_case(ctx, args[0])
+        status = Status[args[1].upper()]
+        if status == Status.CLOSED:
+            raise KeyError  # Mock a KeyError. This command can't close a case.
     except KeyError:
-        return await ctx.redirect(f"No case found for {args[0]!r}.")
-    return await ctx.redirect(f"{case}")
+        return await ctx.reply("Invalid case status provided.")
+    update = await update_single_elem_case_prep(
+        ctx=ctx,
+        case=case,
+        action="Case Status",
+        new_key="status",
+        new_item=status,
+        enum=True,
+    )
+    if update:
+        return await ctx.reply(update)
```

Comment:
this leaves the possibility of a silent failure?

Reply:
Fixed by 7b26b68 in #338. Update will always emit *something*.